src/apps/morpho/utils/morpho-blue/Blue.maths.ts

```diff
@@ -0,0 +1,106 @@
+import { MarketParams, MarketState, UserPosition } from './interfaces';
+
+import { MulDiv, getConvertToAssets, getConvertToShares, mulDivDown, mulDivUp, pow } from 'evm-maths/lib/utils';
+import { wadDivUp, wadMulDown } from 'evm-maths/lib/wad';
+import { WAD } from 'evm-maths/lib/constants';
+
+export const OraclePriceAbi = [
+  {
+    constant: true,
+    inputs: [],
+    name: 'price',
+    outputs: [
+      {
+        name: '',
+        type: 'uint256',
+      },
+    ],
+    payable: false,
+    stateMutability: 'view',
+    type: 'function',
+  },
+];
+
+export class MorphoBlueMath {
+  static MAX_UINT_256: bigint = BigInt('0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff');
+
+  static ORACLE_PRICE_OFFSET = BigInt(36);
+  static ORACLE_PRICE_SCALE = pow(BigInt(1), this.ORACLE_PRICE_OFFSET, BigInt(1));
+  static VIRTUAL_ASSETS: bigint = BigInt(1);
+  static VIRTUAL_SHARES: bigint = BigInt(1e6);
+
+  static mulDivFunction: MulDiv = mulDivDown;
+  static mulDivUp: MulDiv = mulDivUp;
+
+  static convertToAssetsFunction = getConvertToAssets(this.VIRTUAL_ASSETS, this.VIRTUAL_SHARES, this.mulDivFunction);
+  static convertToSharesFunction = getConvertToShares(this.VIRTUAL_ASSETS, this.VIRTUAL_SHARES, this.mulDivFunction);
+
+  static wTaylorCompounded = (x: bigint, n: bigint): bigint => {
+    const firstTerm = x * n;
+    const secondTerm = mulDivDown(firstTerm, firstTerm, BigInt(2) * n * WAD);
+    const thirdTerm = mulDivDown(secondTerm, firstTerm, BigInt(3) * n * WAD);
+    return firstTerm + secondTerm + thirdTerm;
+  };
+
+  /// @dev Calculates the value of `assets` quoted in shares, rounding down.
+  static toSharesDown = (assets: bigint, totalAssets: bigint, totalShares: bigint): bigint => {
+    return this.mulDivFunction(assets, totalShares + this.VIRTUAL_SHARES, totalAssets + this.VIRTUAL_ASSETS);
```

Comment:
```suggestion
    return MorphoBlueMath.mulDivFunction(assets, totalShares + this.VIRTUAL_SHARES, totalAssets + this.VIRTUAL_ASSETS);
```

no need to use `this` when calling static functions and it is actually dangerous because the this context can be injected